walletnode/services/artworkregister/task.go

```diff
@@ -280,11 +372,392 @@ func (task *Task) pastelTopNodes(ctx context.Context) (node.List, error) {
+	}
+
+	return nil
+}
+
+func (task *Task) genRQIdentifiersFiles(ctx context.Context) error {
+	log.Debugf("Connect to %s", task.config.RaptorQServiceAddress)
+	conn, err := task.rqClient.Connect(ctx, task.config.RaptorQServiceAddress)
+	if err != nil {
+		return errors.Errorf("failed to connect to raptorQ service %w", err)
+	}
+	defer conn.Close()
+
+	content, err := task.imageEncodedWithFingerprints.Bytes()
+	if err != nil {
+		return errors.Errorf("failed to read image contents: %w", err)
+	}
+
+	rqService := conn.RaptorQ(&rqnode.Config{
+		RqFilesDir: task.Service.config.RqFilesDir,
+	})
+
+	// FIXME :
+	// - check format of artis block hash should be base58 or not
+	encodeInfo, err := rqService.EncodeInfo(ctx, content, task.config.NumberRQIDSFiles, hex.EncodeToString(task.artistBlockHash), task.Request.ArtistPastelID)
+	if err != nil {
+		return errors.Errorf("failed to generate RaptorQ symbols' identifiers %w", err)
+	}
+
+	files := rq.SymbolIdFiles{}
+	for _, rawSymbolIDFile := range encodeInfo.SymbolIdFiles {
+
+		f, err := task.convertToSymbolIDFile(ctx, rawSymbolIDFile)
+		if err != nil {
+			return errors.Errorf("failed to create rqids file %w", err)
+		}
+
+		files = append(files, f)
+	}
+
+	if len(files) < 1 {
+		return errors.Errorf("nuber of raptorq symbol identifiers files must be greater than 1")
+	}
+	if task.rqids, err = files.FileIdentifers(); err != nil {
+		return errors.Errorf("failed to get rq symbols' identifier file's identifier %w", err)
+	}
+	if err != nil {
+		task.UpdateStatus(StatusErrorGenRaptorQSymbolsFailed)
+		return errors.Errorf("gen RaptorQ symbols' identifiers failed %w", err)
+	}
+
+	task.rqSymbolIDFiles = files
+	task.rqEncodeParams = encodeInfo.EncoderParam
+	return nil
+}
+
+func (task *Task) convertToSymbolIDFile(ctx context.Context, rawFile rqnode.RawSymbolIdFile) (*rq.SymbolIdFile, error) {
+	symbolIDFile := rq.SymbolIdFile{
+		Id:                rawFile.Id,
+		BlockHash:         rawFile.BlockHash,
+		PastelID:          rawFile.PastelID,
+		SymbolIdentifiers: rawFile.SymbolIdentifiers,
+		Signature:         nil,
+	}
+
+	js, err := json.Marshal(&symbolIDFile)
+	if err != nil {
+		return nil, errors.Errorf("failed to marshal identifiers file %w", err)
+	}
+
+	symbolIDFile.Signature, err = task.pastelClient.Sign(ctx, js, task.Request.ArtistPastelID, task.Request.ArtistPastelIDPassphrase, "ed448")
+	if err != nil {
+		return nil, errors.Errorf("failed to sign identifier file %w", err)
+	}
+
+	js, err = json.Marshal(&symbolIDFile)
+	if err != nil {
+		return nil, errors.Errorf("failed to marshal identifiers file with signature %w", err)
+	}
+
+	hasher := sha3.New256()
+	src := bytes.NewReader(js)
+	if _, err := io.Copy(hasher, src); err != nil {
+		return nil, errors.Errorf("failed to hash identifiers file %w", err)
+	}
+	symbolIDFile.FileIdentifer = base58.Encode(hasher.Sum(nil))
+
+	return &symbolIDFile, nil
+}
+
+func (task *Task) createArtTicket(_ context.Context) error {
```

Comment:
context is not used. why we don't remove it. except planing to use it in the future

Reply:
Yes, it was planed for future use.